libobs/obs-audio.c

```diff
@@ -58,9 +58,16 @@ static inline void check_audio_output_source_is_monitoring_device(obs_source_t *
 	    strcmp(s->info.id, "coreaudio_output_capture") == 0) {
 		const char *dev_id = NULL;
 		obs_data_t *settings = obs_source_get_settings(s);
+		bool is_pulse = strcmp(s->info.id, "pulse_output_capture") == 0;
 		if (settings) {
 			dev_id = obs_data_get_string(settings, "device_id");
-			if (strcmp(dev_id, audio->monitoring_device_id) == 0) {
+			bool id_match = strcmp(dev_id, audio->monitoring_device_id) == 0;
+			if (is_pulse) {
+				// pulse may append '.monitor'
```

Comment:
Which code does the appending? Is that based on our implementation of pulse audio? Is that something pulse audio somehow does by itself and is out of our control?

Reply:
It's not enough that it "seems to fix it" - if there is indeed a way for us to control this behaviour in the associated source, then it needs to be fixed _there_ and we should not pile a band-aid on top of that in this code. `libobs` code should have as little platform-specific code as possible, and this is the opposite of that.

Indeed a quick search within the pulse plugin code yields this:

https://github.com/obsproject/obs-studio/blob/0b1229632063a13dfd26cf1cd9dd43431d8c68f6/plugins/linux-pulseaudio/pulse-input.c#L243-L250

So we add the `.monitor` suffix and not Pulse audio itself. Whether it is _necessary_ is another question, but it sounds to me like that wasn't even investigated?